### PR TITLE
BUILD: add support for specific gpu ach with ROCM

### PR DESCRIPTION
**Pull request by akolliasAMD**
## What
Added ability to specify ROCM architecture on configure with --with-rocm-arch
options are:
`all` which does the same behavior as before,
`all-arch-no-native` does all default rocm architectures with the exception of native (to be used if no rocm enabled gpus exist in the system)
`--offload-arch=gfx###` for specifying a specific architecture

## Why ?
This allows specifying a specific architecture. It will fix the https://github.com/openucx/ucc/issues/969 issue

### CODE REVIEW

**Comment by manjugv** [comment]
@akolliasAMD Please also get review from @edgargabriel

**Comment by edgargabriel** [comment]
> @akolliasAMD Please also get review from @edgargabriel

@manjugv this look good to me. I don't think the repo allows me to do a formal review, but we discussed this with @akolliasAMD internally before he filed the PR.

**Review by edgargabriel — APPROVED** [review]
actually, I can do a review :-)

**Review by samnordmann — APPROVED** [review]

**Comment by akolliasAMD** [comment]
Can you give it another thumbs up for the CI to continue?

**Comment by Sergei-Lebedev** [comment]
ok to test

**Review by Sergei-Lebedev — APPROVED** [review]

**Review by samnordmann — APPROVED** [review]

**Comment by akolliasAMD** [comment]
Do you want me to just rebase from master and merge?

**Comment by Sergei-Lebedev** [comment]
> Do you want me to just rebase from master and merge?

yes please

**Comment by akolliasAMD** [comment]
ok rebased. Will wait for tests to pass and merge hopefully

**Pull request merged**
